47.md

```diff
@@ -0,0 +1,112 @@
+NIP-47
+======
+
+Nostr Wallet Connect
+--------------------
+
+`draft` `optional` `author:kiwiidb` `author:bumi` `author:semisol` `author:vitorpamplona`
+
+## Rationale
+
+Paying zaps should be possible without the user needing to open a different app to only pay a Lightning invoice.
+This NIP describes a way for users to control a remote Lightning node or a custodial Lightning wallet. When self-hosting, this setup does not require the user to run their own server, thereby bypassing certain hurdles that are commonly encountered when trying to connect to a Lightning node remotely.
+
+## Terms
+
+* **client**: Nostr app on any platform that wants to pay Lightning invoices
+* **wallet service**: Nostr app that typically runs on an always-on computer (eg. in the cloud or on a Raspberry Pi).
+
+## Events
+
+There are two event kinds:
+- `NIP-47 request`: 23194
+- `NIP-47 response`: 23195
+
+Both the request and response events SHOULD only contain one `p` tag, containing the public key of the **wallet service** if this is a request, and the public key of the **client** if this is a response.
+
+The content is encrypted with [NIP04](https://github.com/nostr-protocol/nips/blob/master/04.md), and is a JSON object. The content depends on the kind.
+
+Request:
+```jsonc
+{
+    "cmd": "pay_invoice", // command, string
+    "data": { // data, object
+        "invoice": "lnbc50n1..." // command-related data
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "status": "ok", // status, "ok" | "error"
```

Comment:
Should we use the `OK` of the base Nostr protocol for success? Why duplicate this here?